presto-main/src/main/java/com/facebook/presto/execution/SqlTaskExecution.java

```diff
@@ -1313,12 +1318,25 @@ private void checkLifespanCompletion(Lifespan lifespan)
             if (lifespan.isTaskWide()) {
                 return; // not a driver group
             }
+
             if (!isNoMoreDriverRunners(lifespan)) {
                 return;
             }
+
+            // do we still have running tasks?
             if (getRemainingDriver(lifespan) != 0) {
                 return;
             }
+
+            // no more output will be created
+            outputBuffer.setNoMorePagesForLifespan(lifespan);
```

Comment:
I find it confusing when `check` methods or methods with a name that sounds like they are read-only methods have side effects like this one where we call `setNoMorePagesForLifespan`, which is a side effect.

Reply:
@nezihyigitbasi : For this methods, it's not a check but notify `outputBuffer` there will be no more pages for the given lifespan, similar to `setNoMorePages`

Reply:
What I mean by a check is, this method `checkLifespanCompletion` is a method with name `check*`, which sounds like a read-only method, but it has a side effect. I don't know it's easy to fix, but I just wanted to bring that up.

Reply:
I see. The name is chosen in consistent with `checkTaskCompletion`. I propose to use this name for now until  we can have any better names for both of them :)